src/wasm-interpreter.h

```diff
@@ -2103,6 +2117,44 @@ template<typename GlobalManager, typename SubType> class ModuleInstanceBase {
     void trap(const char* why) override {
       instance.externalInterface->trap(why);
     }
+
+    // Given a value, wrap it to a smaller given number of bytes.
+    Literal wrapToSmallerSize(Literal value, Index bytes) {
+      if (value.type == Type::i32) {
+        switch (bytes) {
+          case 1: {
+            return value.and_(Literal(int32_t(0xff)));
+          }
+          case 2: {
+            return value.and_(Literal(int32_t(0xffff)));
+          }
+          case 4: {
+            break;
+          }
+          default:
+            WASM_UNREACHABLE("unexpected bytes");
+        }
+      } else {
+        assert(value.type == Type::i64);
+        switch (bytes) {
+          case 1: {
+            return value.and_(Literal(int64_t(0xff)));
+          }
+          case 2: {
+            return value.and_(Literal(int64_t(0xffff)));
+          }
+          case 4: {
+            return value.and_(Literal(int64_t(0xffffffffL)));
```

Comment:
Why do we need the `L` suffix here? Does it affect the wasm build, which uses 32-bit longs?

Reply:
Seems safer to use L here, but it may not be necessary, yeah. I'm just not sure that on 32-bit platforms the C spec doesn't allow a valid compiler to turn `0xffffffff` into an `int`, which means `-1`.

Reply:
Ah, then it would probably be better to use `uint64_t` rather than `int64_t` and use `UL` rather than `L`, although just switching to `uint64_t` should be sufficient.

Reply:
`uint64_t, UL` sound good, done.

(Btw, in practice I don't think `uint64_t` changes things: If the value being cast is an `int(-1)` then with either `uint64_t` or `int64_t` it still turns into 64 bits of `1`s, on both 32 and 64 bit for me locally. However, likely the spec prevents that from being a `-1`. I'm just not sure enough to count on it, unless you are.)